drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,296 @@
+                 uint8_t acc_address,
+                 lsm6ds3_acc_sample_rate_t sample_rate,
+                 lsm6ds3_acc_scale_t acc_scale,
+                 lsm6ds3_acc_bandwith_t acc_bandwith,
+                 uint_8_t gyr_address,
+                 lsm6ds3_gyr_sample_rate_t gyr_sample_rate,
+                 lsm6ds3_gyr_scale_t gyr_scale,
+                 lsm6ds3_gyr_scale_125_t gyr_scale_125)
+{
+    int res = 0;
+    char tmp;
+
+	  dev->i2c = i2c;
+	  dev->acc_address = acc_address;
+	  dev->gyr_address = gyr_address;
+	  dev->acc_pin = acc_pin;
+	  dev->gyr_pin = gyr_pin;
+
+	  i2c_acquire(dev->i2c);
+	  i2c_init_master(i2c, I2C_SPEED_NORMAL);
+	  DEBUG("LSM6DS3: reboot ... ");
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL3_C, LSM6DS3_REG_CTRL3_C_BOOT);
+	  i2c_release(dev->i2c);
+    if (res != 1)
+    {
+        DEBUG("[ FAIL ]\n");
+    }
+    DEBUG("[ OK ]\n");
+
+	  /* configure accelerometer */
+	  /* enable all three axis and set sample rate */
+	  tmp = (LSM6DS3_CTRL9_XL_X_EN
+		      | LSM6DS3_CTRL9_XL_Y_EN
+		      | LSM6DS3_CTRL9_XL_Z_EN);
+	  i2c_acquire(dev->i2c);
+    DEBUG("LSM6DS3: configure accelerometer ... ");
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL9_XL, tmp);
+    res += i2c_write_reg(dev->i2c, dev->acc_address
+                         LSM6DS3_REG_CTRL1_XL, acc_scale);
+    if (res != 3)
+    {
+        DEBUG("[ FAIL ]\n");
+    }
+    DEBUG("[ OK ]\n");
+
+	  /* configure accelerometer data ready pin */
+	  gpio_init_in(acc_pin, GPIO_NOPULL);
+
+	  /* configure temperature */
+    DEBUG("LSM6DS3: configure temperature ... ");
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL4_C, LSM6DS3_CTRL4_C_TEMP_EN);
+    if (res != 4)
+    {
+        DEBUG("[ FAIL ]\n");
+    }
+    DEBUG("[ OK ]\n");
+
+    /* configure gyroscope */
+    DEBUG("LSM6DS3: configure gyroscope ... ");
+    tmp = (LSM6DS3_CTRL10_G_X_EN
+          | LSM6DS3_CTRL10_G_Y_EN
+          | LSM6DS3_CTRL10_G_Z_EN);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL10_XL, tmp);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL2_G, gyr_scale);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL2_G, gyr_scale_125);
+    /* configure gyroscope data ready pin */
+    gpio_init_in(gyr_pin, GPIO_NOPULL);
+    i2c_release(dev->i2c);
+    if (res != 7)
+    {
+        DEBUG("[ FAIL ]\n");
+        return -1;
+    }
+    DEBUG("[ OK ]\n");
+
+    return 0;
+}
+
+int lsm6ds3_read_acc(lsm6ds3_t *dev, lsm6ds3_3d_data_t *data)
+{
+    int res;
+    char tmp;
+
+    i2c_acquire(dev->i2c);
+    i2c_read_reg(dev->12c, dev-acc_address, LSM6DS3_STATUS_REG, &tmp);
+    DEBUG("LSM6DS3: status ... %x\n", tmp);
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
```

Comment:
@PeterKietzmann Okey, I also think it's better to initialize the `<res>` with the first i2c function and increment the value then..